Series of nitpicks (#119)
myyoda/principles-paper: main.tex

\section{Introduction}

Data-driven research depends on specific versions of datasets, software, and computational environments, together with a record of how they were used to produce results.

> We adopt the term \textbf{research object} to denote a collection of data, code, and metadata that together represent the research output as a unit.  % This definition of research object is still a bit problematic to me. While a research object is supposed to be the representation of the research output, but data, code, and metadata seem like mostly are the "inputs" to a research. Additionally, tools such as datalad capture both the inputs and output results of a research, so I think your intended definition of a research object must represent both inputs and outputs of a research.

Reviewer: ```suggestion
We adopt the term \textbf{research object} to denote a collection of data, code, and metadata that together represent the research as a complete unit.
```

It is common for the components of a research object to be stored separately---code in one repository, data on a shared drive, environment setup in a wiki page, provenance in a lab notebook or nowhere at all---and this separation undermines rigor, reproducibility, reusability, and efficiency.
Even when gathered in one place, research objects are often tightly coupled to specific environments, opaque to outside collaborators, and difficult to reuse or redistribute.

> The FAIR principles\cite{wilkinson_fair_2016} and FAIR4RS guidelines\cite{barker_introducing_2022} provide structured guidance for making digital objects findable, accessible, interoperable, and reusable.

Reviewer 2: ```suggestion
The FAIR principles\cite{wilkinson_fair_2016} and FAIR4RS guidelines\cite{barker_introducing_2022} provide structured guidance for making digital objects "Findable, Accessible, Interoperable, and Reusable".
```

> The Workflow Community Initiative (WCI-FW)\cite{wilkinson_applying_2025} mapped these principles % We are assuming that the reader knows findable, accessible, interoperable, and reusable are the FAIR principles, and we are referring to the FAIR principles here.

Reviewer 2: ```suggestion
The Workflow Community Initiative (WCI-FW)\cite{wilkinson_applying_2025} mapped these principles onto computational workflows but explicitly chose not to define new ones, instead treating workflows as hybrid data/software objects.
```

Reviewer 2: suggestion reverts to origina, but i'll still capture @candleindark's comment:

> We are assuming that the reader knows findable, accessible, interoperable, and reusable are the FAIR principles, and we are referring to the FAIR principles here.

\section{Introduction}

Data-driven research depends on specific versions of datasets, software, and computational environments, together with a record of how they were used to produce results.

> We adopt the term \textbf{research object} to denote a collection of data, code, and metadata that together represent the research output as a unit.  % This definition of research object is still a bit problematic to me. While a research object is supposed to be the representation of the research output, but data, code, and metadata seem like mostly are the "inputs" to a research. Additionally, tools such as datalad capture both the inputs and output results of a research, so I think your intended definition of a research object must represent both inputs and outputs of a research.

Reviewer: I think the original sentence reads perfectly fine.

Generically, 'data' is not being split in the original sentence into sourcedata vs. derivatives.

Nor, generically, do we need to, since purely computational/synthetic science may not have sourcedata at all.

So the very notion of 'input'/'output' is a step away from a general definition and towards pre-existing notions of particular types of specific workflow patterns.

Author: > Generically, 'data' is not being split in the original sentence into sourcedata vs. derivatives.

If that's the case, I think we should avoid the word "output" in "research output" and use something else. That word does convey the idea of input/output.

Reviewer: That's fair

\section{Introduction}

> Data-driven research depends on specific versions of datasets, software, and computational environments, together with a record of how they were used to produce results.

Reviewer: "Used" is more generic (they could be 'used' independently), with less emphasis on the interactions between the three items, which I think is the point we try to make here

Reviewer 2: ```suggestion
Data-driven research depends on specific versions of datasets, software, and computational environments, and a record of how they were used together to produce results.
```

We adopt the term \textbf{research object} to denote a collection of data, code, and metadata that together represent the research output as a unit.  % This definition of research object is still a bit problematic to me. While a research object is supposed to be the representation of the research output, but data, code, and metadata seem like mostly are the "inputs" to a research. Additionally, tools such as datalad capture both the inputs and output results of a research, so I think your intended definition of a research object must represent both inputs and outputs of a research.

> It is common for the components of a research object to be stored separately---code in one repository, data on a shared drive, environment setup in a wiki page, provenance in a lab notebook or nowhere at all---and this separation undermines rigor, reproducibility, reusability, and efficiency.

Reviewer: "storage" implies less active interaction with these components; the original "managed" gets this across moreso, but still may not be the best word

"fragmentation" is a particular computer-sciency word that can mean other things, so maybe "separation" is the better word there

Author: The problem I see is more the parallel between the examples given and the verb used. What about "managed separately in different locations"?

Reviewer 2: ```suggestion
It is common for the components of a research object to be organized independently---code in one repository, data on a shared drive, environment setup in a wiki page, provenance in a lab notebook or nowhere at all---and this separation undermines rigor, reproducibility, reusability, and efficiency.
```

Idk I chewed on this a while. This was the best i got

Reviewer: I like it 👍
